### ISSUE TITLE
non-adjacent seconds in cross staff chords

### ISSUE THREAD

**Issue by craigsapp** [rism-ch/verovio#254]
Here is a funny corner case to fix (first one found in real music):

![screen shot 2016-07-20 at 11 17 39 pm](https://cloud.githubusercontent.com/assets/3487289/17012963/30db2c70-4ed0-11e6-8335-7e609df4b2cc.png)

The top note of the first chord in staff 1 (E4) should not be offset to the other side of the stem since the D4 note is on the other staff, and therefore not in the way (and also therefore the ledger line should not be extended).  Similar for the second chord, but also note the ledger lines for each staff should also be considered.

MEI data:

``` xml
<?xml version="1.0" encoding="UTF-8"?>
<?xml-model href="http://music-encoding.org/schema/3.0.0/mei-all.rng" type="application/xml" schematypens="http://relaxng.org/ns/structure/1.0"?>
<?xml-model href="http://music-encoding.org/schema/3.0.0/mei-all.rng" type="application/xml" schematypens="http://purl.oclc.org/dsdl/schematron"?>
<mei xmlns="http://www.music-encoding.org/ns/mei" meiversion="3.0.0">
    <meiHead>
        <fileDesc>
            <titleStmt>
                <title />
            </titleStmt>
            <pubStmt>
                <date>2016-07-20 23:17:32</date>
            </pubStmt>
        </fileDesc>
        <encodingDesc>
            <projectDesc>
                <p>Transcoded from Humdrum with Verovio version 0.9.12-dev-96e36ab</p>
            </projectDesc>
        </encodingDesc>
    </meiHead>
    <music>
        <body>
            <mdiv>
                <score>
                    <scoreDef xml:id="scoredef-000000062196711">
                        <staffGrp xml:id="m-000000068138666" symbol="brace">
                            <staffDef xml:id="staffdef-000000168278881" clef.shape="G" clef.line="2" meter.count="4" meter.unit="4" n="1" lines="5" />
                            <staffDef xml:id="staffdef-000000139246092" clef.shape="F" clef.line="4" meter.count="4" meter.unit="4" n="2" lines="5" />
                        </staffGrp>
                    </scoreDef>
                    <section>
                        <measure xml:id="measure-000000146861364" n="1" right="end">
                            <staff xml:id="staff-000000115534766" n="1">
                                <!-- kern: =1-  2Ej 2Bj/ 2dj 2e  2Gj 2cj 2B 2e 2f  == -->
                                <layer xml:id="layer-000000083853181" n="1">
                                    <chord xml:id="chord-000000141648947" dur="2" stem.dir="up">
                                        <note xml:id="note-000000208241294" staff="2" oct="3" pname="e" accid.ges="n" />
                                        <note xml:id="note-000000157335464" staff="2" oct="3" pname="b" accid.ges="n" />
                                        <note xml:id="note-000000140544023" staff="2" oct="4" pname="d" accid.ges="n" />
                                        <note xml:id="note-000000106134587" oct="4" pname="e" accid.ges="n" />
                                    </chord>
                                    <chord xml:id="chord-000000104088191" dur="2">
                                        <note xml:id="note-000000070055694" staff="2" oct="3" pname="g" accid.ges="n" />
                                        <note xml:id="note-000000175518814" staff="2" oct="4" pname="c" accid.ges="n" />
                                        <note xml:id="note-000000161181143" oct="3" pname="b" accid.ges="n" />
                                        <note xml:id="note-000000135609840" oct="4" pname="e" accid.ges="n" />
                                        <note xml:id="note-000000070187999" oct="4" pname="f" accid.ges="n" />
                                    </chord>
                                </layer>
                            </staff>
                            <staff xml:id="staff-000000038106948" n="2">
                                <!-- kern: =1-  1AA  == -->
                                <layer xml:id="layer-000000036941980" n="1">
                                    <note xml:id="note-000000046337193" dur="1" oct="2" pname="a" accid.ges="n" />
                                </layer>
                            </staff>
                        </measure>
                    </section>
                </score>
            </mdiv>
        </body>
    </music>
</mei>
```

**Comment by donbyrd** [comment]
Bizarre -- especially the 2nd chord, where C4 is on the bottom staff while B3 is on the upper! I can't see why someone would do that; aside from being pointless, it makes the left hand stretch a 10th, which is pretty wide. But you saw this in real music? What music?

**Comment by craigsapp** [comment]
The *first* one was in real music, the second was an artificial stress test.

Here is the source from Beethoven sonata op 26 mvmt 3:

![screen shot 2016-07-21 at 8 09 43 am](https://cloud.githubusercontent.com/assets/3487289/17027857/86a80a02-4f1a-11e6-80d3-4ce3b3ab6704.png)

**Comment by donbyrd** [comment]
That makes sense, thanks.

**Comment by craigsapp** [comment]
Here is a slightly different case to also consider:

![screen shot 2016-07-26 at 9 08 13 pm](https://cloud.githubusercontent.com/assets/3487289/17163543/5dc42102-5375-11e6-99df-9ba0a0b39419.png)

How it is rendered in verovio:

![screen shot 2016-07-26 at 9 10 51 pm](https://cloud.githubusercontent.com/assets/3487289/17163552/785be87e-5375-11e6-9c86-862b26a25f7f.png)

(The two half notes chords should be offset by the width of a note)

MEI test data:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<?xml-model href="http://music-encoding.org/schema/3.0.0/mei-all.rng" type="application/xml" schematypens="http://relaxng.org/ns/structure/1.0"?>
<?xml-model href="http://music-encoding.org/schema/3.0.0/mei-all.rng" type="application/xml" schematypens="http://purl.oclc.org/dsdl/schematron"?>
<mei xmlns="http://www.music-encoding.org/ns/mei" meiversion="3.0.0">
    <meiHead>
        <fileDesc>
            <titleStmt>
                <title />
            </titleStmt>
            <pubStmt>
                <date>2016-07-26 21:09:14</date>
            </pubStmt>
        </fileDesc>
        <encodingDesc>
            <projectDesc>
                <p>Transcoded from Humdrum with Verovio version 0.9.12-dev-a018b75-dirty</p>
            </projectDesc>
        </encodingDesc>
    </meiHead>
    <music>
        <body>
            <mdiv>
                <pages type="raw">
                    <!--Coordinates in MEI axis direction-->
                    <page xml:id="page-000000074668087">
                        <system xml:id="system-000000191798830" system.leftmar="0" system.rightmar="0">
                            <measure xml:id="measure-000000025375141" n="7" right="end">
                                <staff xml:id="staff-000000005098664" n="1">
                                    <layer xml:id="layer-000000173350833" n="1">
                                        <mRest xml:id="mrest-000000151144705" />
                                    </layer>
                                </staff>
                                <staff xml:id="staff-000000041252771" n="2">
                                    <layer xml:id="layer-000000023494571" n="1">
                                        <chord xml:id="chord-000000140007135" dur="2" stem.dir="up">
                                            <note xml:id="note-000000160138059" oct="3" pname="e" accid.ges="n" />
                                            <note xml:id="note-000000135953168" oct="3" pname="g" accid.ges="n" />
                                            <note xml:id="note-000000132461744" oct="3" pname="a" accid.ges="n" />
                                            <note xml:id="note-000000162847464" oct="4" pname="c" accid.ges="s" />
                                        </chord>
                                        <chord xml:id="chord-000000016282597" dur="4" stem.dir="up">
                                            <note xml:id="note-000000194067243" oct="3" pname="f" accid.ges="s" />
                                            <note xml:id="note-000000081529475" oct="3" pname="a" accid.ges="n" />
                                            <note xml:id="note-000000169508664" oct="4" pname="d" accid.ges="n" />
                                        </chord>
                                        <rest xml:id="rest-000000074719678" dur="4" />
                                    </layer>
                                    <layer xml:id="layer-000000052181504" n="2">
                                        <chord xml:id="chord-000000014600565" dur="2" stem.dir="down">
                                            <note xml:id="note-000000005745558" oct="2" pname="d" accid.ges="n" />
                                            <note xml:id="note-000000080421087" oct="2" pname="a" accid.ges="n" />
                                            <note xml:id="note-000000013380495" oct="3" pname="d" accid.ges="n" />
                                        </chord>
                                        <chord xml:id="chord-000000185440823" dur="4" stem.dir="down">
                                            <note xml:id="note-000000113131134" oct="2" pname="d" accid.ges="n" />
                                            <note xml:id="note-000000196754438" oct="2" pname="f" accid.ges="s" />
                                            <note xml:id="note-000000054211585" oct="2" pname="a" accid.ges="n" />
                                            <note xml:id="note-000000075271580" oct="3" pname="d" accid.ges="n" />
                                        </chord>
                                        <rest xml:id="rest-000000178631857" dur="4" />
                                    </layer>
                                </staff>
                            </measure>
                        </system>
                    </page>
                </pages>
            </mdiv>
        </body>
    </music>
</mei>
```

The above uses two layers for the bottom staff.  A closer encoding would be to have a cross-staff chord from the top staff:

![screen shot 2016-07-26 at 9 15 48 pm](https://cloud.githubusercontent.com/assets/3487289/17163643/4271fd9c-5376-11e6-9a48-ff2c857b568a.png)

(how to unattach the stem from staff 1?)

MEI data:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<?xml-model href="http://music-encoding.org/schema/3.0.0/mei-all.rng" type="application/xml" schematypens="http://relaxng.org/ns/structure/1.0"?>
<?xml-model href="http://music-encoding.org/schema/3.0.0/mei-all.rng" type="application/xml" schematypens="http://purl.oclc.org/dsdl/schematron"?>
<mei xmlns="http://www.music-encoding.org/ns/mei" meiversion="3.0.0">
    <meiHead>
        <fileDesc>
            <titleStmt>
                <title />
            </titleStmt>
            <pubStmt>
                <date>2016-07-26 21:15:32</date>
            </pubStmt>
        </fileDesc>
        <encodingDesc>
            <projectDesc>
                <p>Transcoded from Humdrum with Verovio version 0.9.12-dev-a018b75-dirty</p>
            </projectDesc>
        </encodingDesc>
    </meiHead>
    <music>
        <body>
            <mdiv>
                <pages type="raw">
                    <!--Coordinates in MEI axis direction-->
                    <page xml:id="page-000000187697759">
                        <system xml:id="system-000000148247926" system.leftmar="0" system.rightmar="0">
                            <measure xml:id="measure-000000144265402" n="7" right="end">
                                <staff xml:id="staff-000000148583939" n="1">
                                    <layer xml:id="layer-000000131123635" n="1">
                                        <chord xml:id="chord-000000045489743" dur="2" stem.dir="up">
                                            <note xml:id="note-000000097605786" staff="2" oct="3" pname="e" accid.ges="n" />
                                            <note xml:id="note-000000015922968" staff="2" oct="3" pname="g" accid.ges="n" />
                                            <note xml:id="note-000000062889695" staff="2" oct="3" pname="a" accid.ges="n" />
                                            <note xml:id="note-000000100006697" staff="2" oct="4" pname="c" accid.ges="s" />
                                        </chord>
                                        <chord xml:id="chord-000000050634342" dur="4" stem.dir="up">
                                            <note xml:id="note-000000057964310" staff="2" oct="3" pname="f" accid.ges="s" />
                                            <note xml:id="note-000000173704845" staff="2" oct="3" pname="a" accid.ges="n" />
                                            <note xml:id="note-000000153034036" staff="2" oct="4" pname="d" accid.ges="n" />
                                        </chord>
                                        <rest xml:id="rest-000000166685236" dur="4" staff="2" />
                                    </layer>
                                </staff>
                                <staff xml:id="staff-000000077244538" n="2">
                                    <layer xml:id="layer-000000002529457" n="1">
                                        <chord xml:id="chord-000000091801881" dur="2" stem.dir="down">
                                            <note xml:id="note-000000053708324" oct="2" pname="d" accid.ges="n" />
                                            <note xml:id="note-000000160214021" oct="2" pname="a" accid.ges="n" />
                                            <note xml:id="note-000000148325410" oct="3" pname="d" accid.ges="n" />
                                        </chord>
                                        <chord xml:id="chord-000000157037759" dur="4" stem.dir="down">
                                            <note xml:id="note-000000127754411" oct="2" pname="d" accid.ges="n" />
                                            <note xml:id="note-000000027041382" oct="2" pname="f" accid.ges="s" />
                                            <note xml:id="note-000000104104452" oct="2" pname="a" accid.ges="n" />
                                            <note xml:id="note-000000047010043" oct="3" pname="d" accid.ges="n" />
                                        </chord>
                                        <rest xml:id="rest-000000109370377" dur="4" />
                                    </layer>
                                </staff>
                            </measure>
                        </system>
                    </page>
                </pages>
            </mdiv>
        </body>
    </music>
</mei>
```

**Comment by lpugin** [comment]
The first case is now fixed. The layer overlap will be fixed together with #96 

![image](https://cloud.githubusercontent.com/assets/689412/25199388/f713b56e-254a-11e7-846a-a8d8b8fdb83e.png)
